File: local-test-setup/story-writer-agent/frontend/src/app/story/story.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface StoryRequest {
  title: string;
  recipient_email: string;
}

export interface StoryResponse {
  title: string;
  story: string;
  db_status: string;
  db_tool_used: string | null;
  email_status: string;
  email_tool_used: string | null;
}

@Injectable({ providedIn: 'root' })
export class StoryService {
  constructor(private http: HttpClient) {}

  createStory(request: StoryRequest): Observable<StoryResponse> {
    return this.http.post<StoryResponse>(`${environment.apiBaseUrl}/stories`, request);
  }
}

File: local-test-setup/story-writer-agent/backend/app/services/story_service.py
from app.agent.graph import story_graph
from app.schemas import StoryResponse


async def run_story_pipeline(title: str, recipient_email: str) -> StoryResponse:
    result = await story_graph.ainvoke({"title": title, "recipient_email": recipient_email})
    return StoryResponse(
        title=title,
        story=result["story"],
        db_status=result["db_status"],
        db_tool_used=result.get("db_tool_used"),
        email_status=result["email_status"],
        email_tool_used=result.get("email_tool_used"),
    )
